pkg/node_reputation/node_reputation.go

```diff
@@ -0,0 +1,282 @@
+			switch {
+			case row.timestamp > other.timestamp && row.nodeName == other.nodeName:
+				rowEndian.WriteString("1")
+				otherEndian.WriteString("0")
+			case row.timestamp < other.timestamp && row.nodeName == other.nodeName:
+				rowEndian.WriteString("0")
+				otherEndian.WriteString("1")
+			default:
+				rowEndian.WriteString("0")
+				otherEndian.WriteString("0")
+			}
+
+		case shardsModifiedColumn:
+			if row.shardsModified > 0 {
+				rowEndian.WriteString("0")
+			} else {
+				rowEndian.WriteString("1")
+			}
+			if other.shardsModified > 0 {
+				otherEndian.WriteString("0")
+			} else {
+				otherEndian.WriteString("1")
+			}
+
+		case falseClaimsColumn:
+			switch {
+			case row.falseClaims < other.falseClaims:
+				rowEndian.WriteString("1")
+				otherEndian.WriteString("0")
+			case row.falseClaims > other.falseClaims:
+				rowEndian.WriteString("0")
+				otherEndian.WriteString("1")
+			default:
+				rowEndian.WriteString("0")
+				otherEndian.WriteString("0")
+			}
+
+		case auditSuccessColumn:
+			switch {
+			case row.auditSuccessRatio() > other.auditSuccessRatio():
+				rowEndian.WriteString("1")
+				otherEndian.WriteString("0")
+			case row.auditSuccessRatio() < other.auditSuccessRatio():
+				rowEndian.WriteString("0")
+				otherEndian.WriteString("1")
+			default:
+				rowEndian.WriteString("0")
+				otherEndian.WriteString("0")
+			}
+
+		case uptimeColumn:
+			switch {
+			case row.uptime > other.uptime:
+				rowEndian.WriteString("1")
+				otherEndian.WriteString("0")
+			case row.uptime < other.uptime:
+				rowEndian.WriteString("0")
+				otherEndian.WriteString("1")
+			default:
+				rowEndian.WriteString("0")
+				otherEndian.WriteString("0")
+			}
+
+		case latencyColumn:
+			switch {
+			case row.latency < other.latency:
+				rowEndian.WriteString("1")
+				otherEndian.WriteString("0")
+			case row.latency > other.latency:
+				rowEndian.WriteString("0")
+				otherEndian.WriteString("1")
+			default:
+				rowEndian.WriteString("0")
+				otherEndian.WriteString("0")
+			}
+
+		case amountOfDataStoredColumn:
+			switch {
+			case row.amountOfDataStored > other.amountOfDataStored:
+				rowEndian.WriteString("1")
+				otherEndian.WriteString("0")
+			case row.amountOfDataStored < other.amountOfDataStored:
+				rowEndian.WriteString("0")
+				otherEndian.WriteString("1")
+			default:
+				rowEndian.WriteString("0")
+				otherEndian.WriteString("0")
+			}
+		}
+	}
+
+	var res nodeReputationRecord
+
+	if rowEndian.String() > otherEndian.String() {
+		res = row
+	} else {
+		res = other
+	}
+
+	//side effects
+	/*
+		fmt.Printf("endian: %v, me: %v\n", rowEndian.String(), row.nodeName)
+		fmt.Printf("endian: %v, other: %v\n", otherEndian.String(), other.nodeName)
+
+		fmt.Printf("WINNER: %v\n\n", res)
+	*/
+	return res
+}
+
+// serde converts private record to public reputation record
+func (row nodeReputationRecord) serde(score float32) proto.NodeReputationRecord {
+	return proto.NodeReputationRecord{
```

Comment:
maybe we should just use proto.NodeReputationRecord everywhere instead of having to manually copy all the fields?